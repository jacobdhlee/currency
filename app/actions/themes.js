export const PRIMARY_COLOR_CHANGE = 'PRIMARY_COLOR_CHANGE';

export const changeTheme = color => ({
  type: PRIMARY_COLOR_CHANGE,
  color,
});
