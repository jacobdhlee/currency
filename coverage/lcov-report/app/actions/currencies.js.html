<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/actions/currencies.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">app/actions</a> currencies.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/12</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js"><span class="cstat-no" title="statement not covered" >export const SWAP_CURRENCY = 'SWAP_CURRENCY';<span class="cstat-no" title="statement not covered" ></span></span>
export const CHANGE_CURRENC<span class="cstat-no" title="statement not covered" >Y_AMOUNT = 'CHANGE_CURRENCY_AMOUNT';</span>
export const CHANGE_BASE_<span class="cstat-no" title="statement not covered" >CURRENCY = 'CHANGE_BASE_CURRENCY';</span>
export const CHANGE_QUOTE_<span class="cstat-no" title="statement not covered" >CURRENCY = 'CHANGE_QUOTE_CURRENCY';</span>
export const GET_INITIAL_CO<span class="cstat-no" title="statement not covered" >NVERSION = 'GET_INITIAL_CONVERSION';</span>
export const CONVERSIO<span class="cstat-no" title="statement not covered" >N_RESULT = 'CONVERSION_RESULT';</span>
export const CONVERSI<span class="cstat-no" title="statement not covered" >ON_ERROR = 'CONVERSION_ERROR';</span>
&nbsp;
export const swa<span class="cstat-no" title="statement not covered" >pCurrecy = () =&gt; ({<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span></span>
  type: SWAP_CURRENCY,
});
&nbsp;
export const changeCurren<span class="cstat-no" title="statement not covered" >cyAmount = amount =&gt; ({<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span></span>
  type: CHANGE_CURRENCY_AMOUNT,
  amount: parseFloat(amount),
});
&nbsp;
export const chageBase<span class="cstat-no" title="statement not covered" >Currency = currency =&gt; ({<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span></span>
  type: CHANGE_BASE_CURRENCY,
  currency,
});
&nbsp;
export const chageQuote<span class="cstat-no" title="statement not covered" >Currency = currency =&gt; ({<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span></span>
  type: CHANGE_QUOTE_CURRENCY,
  currency,
});
&nbsp;
export const getInitialCo<span class="cstat-no" title="statement not covered" >nversion = () =&gt; ({<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span></span>
  type: GET_INITIAL_CONVERSION,
});
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Dec 16 2017 14:22:19 GMT-0500 (EST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
