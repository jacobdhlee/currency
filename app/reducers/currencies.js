import { SWAP_CURRENCY, CHANGE_CURRENCY_AMOUNT } from '../actions/currencies';

const INITIAL_STATE = {
  baseCurrency: 'USD',
  quoteCurrency: 'KRW',
  amount: 100,
  conversion: {},
};

const reducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case SWAP_CURRENCY:
      return {
        ...state,
        baseCurrency: state.quoteCurrency,
        quoteCurrency: state.baseCurrency,
      };
    case CHANGE_CURRENCY_AMOUNT:
      return {
        ...state,
        amount: action.amount || 0,
      };
    default:
      return state;
  }
};

export default reducer;
