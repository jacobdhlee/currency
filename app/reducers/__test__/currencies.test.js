import reducer from '../currencies';
import { getInitialConversion } from '../../actions/currencies';

it('set initial state', () => {
  const expected = {
    baseCurrency: 'USD',
    quoteCurrency: 'GBP',
    amount: 100,
    conversions: {},
    error: null,
  };
  const actual = reducer(undefined, {});
  expect(actual).toEqual(expected);
});

it('set nested data on initial fetch', () => {
  const expected = {
    baseCurrency: 'USD',
    quoteCurrency: 'GBP',
    amount: 100,
    conversions: {
      USD: {
        isFetching: true,
        date: '',
        rates: {},
      },
    },
    error: null,
  };
  const actual = reducer(undefined, getInitialConversion());
  expect(actual).toEqual(expected);
});

// snapshot version of testing

it('set initial state', () => {
  const actual = reducer(undefined, {});
  expect(actual).toMatchSnapshot();
});

it('set nested data on initial fetch', () => {
  const actual = reducer(undefined, getInitialConversion());
  expect(actual).toMatchSnapshot();
});
