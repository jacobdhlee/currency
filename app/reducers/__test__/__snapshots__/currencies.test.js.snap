// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`set initial state 1`] = `
Object {
  "amount": 100,
  "baseCurrency": "USD",
  "conversions": Object {},
  "error": null,
  "quoteCurrency": "GBP",
}
`;

exports[`set nested data on initial fetch 1`] = `
Object {
  "amount": 100,
  "baseCurrency": "USD",
  "conversions": Object {
    "USD": Object {
      "date": "",
      "isFetching": true,
      "rates": Object {},
    },
  },
  "error": null,
  "quoteCurrency": "GBP",
}
`;
